The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/6116861613/job/16602657006) [(plain)](https://github.com/rust-lang-ci/rust/commit/8f0fd01e5a46f6d038d8d390f98d645a24f644be/checks/16602657006/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
     |
1531 |         Stdio::from_inner(inherit.into())
     |                                   ^^^^ the trait `core::convert::From<io::stdio::Stdout>` is not implemented for `sys::wasm::process::Stdio`
     |
     = help: the following other types implement trait `core::convert::From<T>`:
               <sys::wasm::process::Stdio as core::convert::From<sys::wasm::fs::File>>
               <sys::wasm::process::Stdio as core::convert::From<AnonPipe>>
     = note: required for `io::stdio::Stdout` to implement `core::convert::Into<sys::wasm::process::Stdio>`

error[E0277]: the trait bound `sys::wasm::process::Stdio: core::convert::From<io::stdio::Stderr>` is not satisfied
     |
1560 |         Stdio::from_inner(inherit.into())
1560 |         Stdio::from_inner(inherit.into())
     |                                   ^^^^ the trait `core::convert::From<io::stdio::Stderr>` is not implemented for `sys::wasm::process::Stdio`
     = help: the following other types implement trait `core::convert::From<T>`:
     = help: the following other types implement trait `core::convert::From<T>`:
               <sys::wasm::process::Stdio as core::convert::From<sys::wasm::fs::File>>
               <sys::wasm::process::Stdio as core::convert::From<AnonPipe>>
     = note: required for `io::stdio::Stderr` to implement `core::convert::Into<sys::wasm::process::Stdio>`
For more information about this error, try `rustc --explain E0277`.
[RUSTC-TIMING] std test:false 1.042
warning: `std` (lib) generated 1 warning
error: could not compile `std` (lib) due to 2 previous errors; 1 warning emitted
```

</details>

--- a/library/std/src/sys/unsupported/process.rs
+++ b/library/std/src/sys/unsupported/process.rs
@@ -27,6 +27,8 @@ pub struct StdioPipes {
     pub stderr: Option<AnonPipe>,
 }
 
+// FIXME: This should be a unit struct, so we can always construct it
+// The value here should be never used, since we cannot spawn processes.
 pub enum Stdio {
     Inherit,
     Null,
@@ -87,8 +89,26 @@ impl From<AnonPipe> for Stdio {
     }
 }
 
+impl From<io::Stdout> for Stdio {
+    fn from(_: io::Stdout) -> Stdio {
+        // FIXME: This is wrong.
+        // Instead, the Stdio we have here should be a unit struct.
+        panic!("unsupported")
+    }
+}
+
+impl From<io::Stderr> for Stdio {
+    fn from(_: io::Stderr) -> Stdio {
+        // FIXME: This is wrong.
+        // Instead, the Stdio we have here should be a unit struct.
+        panic!("unsupported")
+    }
+}
+
 impl From<File> for Stdio {
     fn from(_file: File) -> Stdio {
+        // FIXME: This is wrong.
+        // Instead, the Stdio we have here should be a unit struct.
         panic!("unsupported")
     }
 }

